src/main/java/dev/jbang/source/Project.java

```diff
@@ -26,6 +28,10 @@
 public class Project {
 	@Nonnull
 	private final ResourceRef resourceRef;
+
+	@Nonnull
```

Comment:
We'll have to come up with a different solution for this because Project is a model object and cannot contain anything that is not directly related top project management.

Reply:
I haven't looked at it in-depth yet, but I'm guessing this would need to be reversed, perhaps have some kind of factory that given a Project can return a RootContext.